app/main.py

```diff
@@ -1,3 +1,105 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __add__(self, other: "Distance") -> "Distance":
+        if not isinstance(other, Distance):
+            if isinstance(other, int | float):
+                return Distance(self.km + other)
+            raise TypeError(
```

Comment:
You have a lot of duplicate code for checking, try to put is in separate method.